nomad/structs/structs.go

```diff
@@ -5553,6 +5568,40 @@ func (t *Task) Warnings() error {
 	return mErr.ErrorOrNil()
 }
 
+type TaskKind string
+
+const connectPrefix = "connect-proxy:"
```

Comment:
I'm referencing a similar const in the `github.com/hashicorp/nomad/nomad` package in my admission controller branch. I wonder if we should export this to reduce code duplication?

If we do I'd request we drop the `:` because that prefix is also used when naming the task and port labels ex. `connect-proxy-<service>`. See https://github.com/hashicorp/nomad/pull/6116/files#diff-634941deff286e28e042a781a7e10764R10-R14

Reply:
Ha, I left this comment and deleted it because I figured I'd just export it.

But yeah, I think we'll want to export this, drop the `:`, and maybe toss a method on TaskKind to do the splitting for us.

It can either happen in this PR or after.